### DISCUSSION TITLE
Throttling based on users

### DISCUSSION THREAD

**Discussion by lewoudar** [eadwinCode/django-ninja-extra#172] [Ideas]
Hi everyone,
I'm thinking of a way to improve the [throttling](https://eadwincode.github.io/django-ninja-extra/tutorial/throttling/) feature. It would be good if this feature could also be based on user characteristics. In many APIs we often want to have different rate limits for different types of clients. Actually, I don't think this is possible with Django Ninja Extra.

I see that in Django Ninja, we can create a [custom throttling](https://django-ninja.dev/guides/throttling/#custom-throttles) class, this is perhaps an option for me.

Also, is it possible to use the [decorator feature](https://django-ninja.dev/whatsnew_v1/) (at the bottom of the page) of Django Ninja? This will allow me to use a third-party package such as [django-ratelimit](https://django-ratelimit.readthedocs.io/en/latest/index.html).

Thanks in advance for the answers.

**Comment by eadwinCode** [comment]
Django Ninja throttling copied NinjaExtra Throttling implementation but with different usage.  So what you can do with Django Ninja throttling is also applicable to NinjaExtra throttling. They are all the same. Its just your choice on which you want to use. 

I plan to deprecate throttling in NinjaExtra since the same codebase now exist in Django Ninja.

**Reply by eadwinCode** [reply]
>Also, is it possible to use the [decorator feature](https://django-ninja.dev/whatsnew_v1/) (at the bottom of the page) of Django Ninja?

Yes you can

**Reply by lewoudar** [reply]
oh nice! Thank you for your answers.
